lib/temporal/metadata.rb

```diff
@@ -43,12 +20,12 @@ def activity_metadata_from(task, namespace)
           workflow_run_id: task.workflow_execution.run_id,
           workflow_id: task.workflow_execution.workflow_id,
           workflow_name: task.workflow_type.name,
-          headers: headers(task.header&.fields),
+          headers: from_payload_map(task.header&.fields || {}),
           heartbeat_details: from_details_payloads(task.heartbeat_details)
         )
       end
 
-      def workflow_task_metadata_from(task, namespace)
+      def generate_workflow_task_metadata(task, namespace=nil)
```

Comment:
When do we expect a `namespace` to be `nil`?

Reply:
Verified all call sites specify a value and removed the default